ch6/03.scm

```diff
@@ -2,144 +2,60 @@
 
 ;; 1. (lambda (x y) (p (+ 8 x) (q y)))
 ;; ==>
-(lambda (x y cont) (q y (lambda(v)
-			  (p v (cont (+ 8 x cont))))))
+(lambda (x y cont)
+  (q y (lambda (val)
+	 ((lambda (val2) (p val2 val cont))
+	  (+ 8 x)))))
 
 ;;2. (lambda (x y u v) (+ 1 (f (g x y) (+ u v))))
 ;; ==>
-
 (lambda (x y u v cont)
-  (f ((lambda (v1, v2)
-	(cont (+ 1 v1 v2)))
-      (g x y)
-      (+ u v))))
-
-(lambda (x y u v cont)
-  (g x y (lambda (v1)
-	   (f ((lambda (v1, v2)
-		 (cont (+ 1 v1 v2)))
-	       v1)))
-	       (+ u v)))
-
-
-(lambda (x y u v cont)
-  (g x y (f ((lambda (v1, v2)
-                 (cont (+ 1 v1 v2)))
-	     (+ u v)))))
-
-
-(lambda (x y u v cont)
-  (g x y (f ((lambda (v1)
-               (cont (+ 1 v1 (+ u v))))))))
+  (g x y (lambda (val)
+	   ((lambda (val2)
+	      ((lambda (val3) (cont (+ 1 val3)))
+	       (f val val2)))
+	    (+ u v)))))
```

Comment:
more simpler?
```scheme
(lambda (x y u v cont)
  (g x y (lambda(v2)
           (f v2 (+ u v)
              (lambda (v1)
                (cont (+ 1 v1)))
              ))))
```

from the example in page 199, 
and The CPS Recipe on page 200.